contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,334 @@
+        resetGame(game);
+        balances[playerOne] = balances[playerOne].add(forfeit);
+
+        emit ForfeitPaid(gameKey, playerOne, forfeit);
+    }
+
+    function playerTwoCollectsForfeit(bytes32 gameKey) whenNotPaused external whenNotPaused {
+
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_TWO_MOVE, INVALID_STEP_MSG);
+        address playerTwo = game.playerTwo;
+        require(msg.sender == playerTwo, INVALID_PLAYER_MSG);
+        require(block.timestamp >= game.expiryDate, GAME_NOT_EXPIRED_MSG);
+
+        uint256 forfeit = game.stake.mul(2);
+        resetGame(game);
+        balances[playerTwo] = balances[playerTwo].add(forfeit);
+
+        emit ForfeitPaid(gameKey, playerTwo, forfeit);
+    }
+
+    function playerOneEndsGame(bytes32 gameKey) whenNotPaused external whenNotPaused {
+
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_MOVED, INVALID_STEP_MSG);
+        address playerOne = game.playerOne;
+        require(msg.sender == playerOne, INVALID_PLAYER_MSG);
+
+        uint256 stake = game.stake;
+        resetGame(game);
+        balances[playerOne] = balances[playerOne].add(stake);
+
+        emit PlayerOneEndsGame(gameKey, playerOne, stake);
+    }
+
+    function withdraw(uint amount) external whenNotPaused returns(bool success) {
+
+        uint256 withdrawerBalance = balances[msg.sender];
+        require(amount > 0, "The value must be greater than 0");
+
+        balances[msg.sender] = withdrawerBalance.sub(amount);
+        emit WithDraw(msg.sender, amount);
+
+        (success, ) = msg.sender.call{value: amount}("");
+        require(success, "Transfer failed");
+    }
+
+    // We don't want to delete the struct so the key cannot be reused
+    function resetGame(Game storage game) internal {
+
+        game.stake = 0;
+        game.playerTwo = address(0);
+        game.playerTwoMoveHash = bytes32(0);
+        game.playerOneMove = Moves.EMPTY;
+        game.expiryDate = 0;
+        game.step = Steps.UNSTARTED;
+    }
+
+    // This can only be used once because it is used as a mapping key
+    function createPlayerOneMoveHash(address player, bytes32 secret, Moves move) public pure returns(bytes32) {
+
+        require(secret != bytes32(0), SECRET_EMPTY_MSG);
+        require(Moves(move) > Moves.EMPTY, INVALID_MOVE_MSG);
+
+        return keccak256(abi.encodePacked(player, secret, uint256(move)));
+    }
+
+    // Even if the secret is reused, the hash will always be unique because the game key will be unique
+    function createPlayerTwoMoveHash(address player, bytes32 gameKey, bytes32 secret, Moves move) public pure returns (bytes32) {
+
+        require(gameKey != bytes32(0), "Game key cannot be empty");
+        require(secret != bytes32(0), SECRET_EMPTY_MSG);
+        require(Moves(move) > Moves.EMPTY, INVALID_MOVE_MSG);
+
+        return keccak256(abi.encodePacked(player, gameKey, secret, uint256(move)));
```

Comment:
What `encodePacked` does is pack tightly. So by including `uint256(move)` you are telling it to pad-left with zeroes. That makes it easier I imagine to confirm with `soliditySha3`.
On the other hand, it costs a wee bit more gas for no extra safety, since you only add zeroes.
Want to try `encodePacked(..., move)`? And adjust `soliditySha3(..., { type: "uint8", value: Rock })`.

Reply:
This was exactly my reason. Thank you for the advice. I will try that out.